x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -43,15 +43,15 @@ func (k Keeper) GetDataCommitmentForHeight(ctx sdk.Context, height uint64) (type
 			height,
 		)
 	}
-	latestNonce := k.GetLatestAttestationNonce(ctx)
-	for i := uint64(0); i < latestNonce; i++ {
+	latestNonce := int(k.GetLatestAttestationNonce(ctx))
+	for i := latestNonce; i >= 0; i-- {
```

Comment:
```suggestion
	for i := latestNonce; i > 0; i-- {
```
Because there is no attestation 0

Reply:
oh lol then I can revert and use `uints`, sorry for the back and forth

Reply:
Thanks for handling this :D :D